projects/ad7616_sdz/common/ad7616_bd.tcl

```diff
@@ -2,87 +2,177 @@
 ## Copyright (C) 2019-2023 Analog Devices, Inc. All rights reserved.
 ### SPDX short identifier: ADIBSD
 ###############################################################################
-
-# system level parameters
-set SI_OR_PI $ad_project_params(SI_OR_PI)
-
-puts "build parameters: SI_OR_PI: $SI_OR_PI"
-
-# data interfaces
-
-create_bd_port -dir O rx_sclk
-create_bd_port -dir O rx_sdo
-create_bd_port -dir I -from 1 -to 0 rx_sdi
-
-create_bd_port -dir O -from 15 -to 0 rx_db_o
-create_bd_port -dir I -from 15 -to 0 rx_db_i
-create_bd_port -dir O rx_db_t
-create_bd_port -dir O rx_rd_n
-create_bd_port -dir O rx_wr_n
+##--------------------------------------------------------------
+
+# IMPORTANT: Set AD7616 operation and interface mode
+#
+# The get_env_param procedure retrieves parameter value from the environment if exists,
+# other case returns the default value specified in its second parameter field.
+#
+#   How to use over-writable parameters from the environment:
+#
+#    e.g.
+#      make SER_PAR_N=0
+#
+#    SER_PAR_N  - Defines the interface type (serial OR parallel)
+#               - Default value is 1
+#
+# LEGEND: Serial    - 1
+#         Parallel  - 0
+#
+# NOTE : This switch is a 'hardware' switch. Please rebuild the design if the
+# variable has been changed.
+#     SL5 - mounted - Serial
+#     SL5 - unmounted - Parallel
+#
+##--------------------------------------------------------------
+
+set SER_PAR_N $ad_project_params(SER_PAR_N)
+puts "build parameters: SER_PAR_N: $SER_PAR_N"
 
 # control lines
 
 create_bd_port -dir O rx_cnvst
-create_bd_port -dir O rx_cs_n
 create_bd_port -dir I rx_busy
 
 # instantiation
 
-ad_ip_instance axi_ad7616 axi_ad7616
-ad_ip_parameter axi_ad7616 CONFIG.IF_TYPE $SI_OR_PI
+# dma
 
 ad_ip_instance axi_dmac axi_ad7616_dma
-ad_ip_parameter axi_ad7616_dma CONFIG.DMA_TYPE_SRC 2
 ad_ip_parameter axi_ad7616_dma CONFIG.DMA_TYPE_DEST 0
 ad_ip_parameter axi_ad7616_dma CONFIG.CYCLIC 0
 ad_ip_parameter axi_ad7616_dma CONFIG.DMA_2D_TRANSFER 0
-ad_ip_parameter axi_ad7616_dma CONFIG.DMA_DATA_WIDTH_SRC 16
 ad_ip_parameter axi_ad7616_dma CONFIG.DMA_DATA_WIDTH_DEST 64
 
-# interface connections
-if {$SI_OR_PI == 0} {
+# axi_pwm_gen
 
-  ad_connect  rx_sclk axi_ad7616/rx_sclk
-  ad_connect  rx_sdo axi_ad7616/rx_sdo
-  ad_connect  rx_sdi axi_ad7616/rx_sdi
-  ad_connect  rx_cs_n axi_ad7616/rx_cs_n
+ad_ip_instance axi_pwm_gen ad7616_pwm_gen
```

Comment:
I think that ad7616_pwm_gen and spi_clkgen instantiation should be interface-independent because they are used in parallel mode also.